src/BullaFrendLend.sol

```diff
@@ -0,0 +1,213 @@
+     */
+    function getLoan(uint256 claimId) external view returns (Loan memory) {
+        Claim memory claim = _bullaClaim.getClaim(claimId);
+        _checkController(claim.controller);
+        
+        LoanDetails memory loanDetails = _loanDetailsByClaimId[claimId];
+        
+        return Loan({
+            claimAmount: claim.claimAmount,
+            paidAmount: claim.paidAmount,
+            status: claim.status,
+            binding: claim.binding,
+            payerReceivesClaimOnPayment: claim.payerReceivesClaimOnPayment,
+            debtor: claim.debtor,
+            token: claim.token,
+            controller: claim.controller,
+            dueBy: loanDetails.dueBy,
+            interestBPS: loanDetails.interestBPS,
+            termLength: loanDetails.termLength
+        });
+    }
+
+    /**
+     * @notice Allows a user to create and offer a loan to a potential debtor
+     * @param offer The loan offer parameters
+     * @return The ID of the created loan offer
+     */
+    function offerLoan(LoanOffer calldata offer) external payable returns (uint256) {
+        if (msg.value != fee) revert IncorrectFee();
+        if (msg.sender != offer.creditor) revert NotCreditor();
+        if (offer.termLength == 0) revert InvalidTermLength();
+        if (offer.token == address(0)) revert NativeTokenNotSupported();
+
+        uint256 offerId = ++loanOfferCount;
+        loanOffers[offerId] = offer;
+
+        emit LoanOffered(offerId, msg.sender, offer, block.timestamp);
+
+        return offerId;
+    }
+
+    /**
+     * @notice Allows a debtor or creditor to reject or rescind a loan offer
+     * @param offerId The ID of the loan offer to reject
+     */
+    function rejectLoanOffer(uint256 offerId) external {
+        LoanOffer memory offer = loanOffers[offerId];
+        
+        if (offer.creditor == address(0)) revert LoanOfferNotFound();
+        if (msg.sender != offer.creditor && msg.sender != offer.debtor) revert NotCreditorOrDebtor();
+
+        delete loanOffers[offerId];
+
+        emit LoanOfferRejected(offerId, msg.sender, block.timestamp);
+    }
+
+    /**
+     * @notice Allows a debtor to accept a loan offer and receive payment
+     * @param offerId The ID of the loan offer to accept
+     * @param metadata Optional metadata for the claim
+     * @return The ID of the created claim
+     */
+    function acceptLoan(uint256 offerId, ClaimMetadata calldata metadata) external returns (uint256) {
+        LoanOffer memory offer = loanOffers[offerId];
+        
+        if (offer.creditor == address(0)) revert LoanOfferNotFound();
+        if (msg.sender != offer.debtor) revert NotDebtor();
+
+        delete loanOffers[offerId];
+
+        uint256 claimAmount = offer.loanAmount + (offer.loanAmount * offer.interestBPS) / MAX_BPS;
+
+        CreateClaimParams memory claimParams = CreateClaimParams({
+            creditor: offer.creditor,
+            debtor: offer.debtor,
+            claimAmount: claimAmount,
```

Comment:
I know you are directly porting the contracts over, so wouldn't address this comment in this PR, but we will want claimAmount to be the pricinpal, the interest component to be basically calculated every second (we haven't discussed this explicitly but it seems to be the easiest way to do it).

Reply:
we can configure this